### PR TITLE
Russian translation

### PR DESCRIPTION
**Pull request by Sturmlilie**
Translation by `kazmurenko` on CurseForge.

### CODE REVIEW

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 1]
```diff
@@ -1,59 +1,59 @@
...
-    "item.tis3d.module_random_access_memory": "RAM Module",
-    "item.tis3d.module_read_only_memory": "ROM Module",
-    "item.tis3d.module_redstone": "Redstone Module",
-    "item.tis3d.module_sequencer": "Sequencer Module",
-    "item.tis3d.module_serial_port": "Serial Port Module",
-    "item.tis3d.module_stack": "Stack Module",
-    "item.tis3d.module_terminal": "Terminal Module",
-    "item.tis3d.module_timer": "Timer Module",
-    "item.tis3d.prism": "Prism",
+    "item.tis3d.book_code": "Код библия",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.book_code": "Библия Кода",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 2]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.book_manual": "TIS-3D Справочное руководство",
+    "item.tis3d.key": "Ключ",
+    "item.tis3d.key_creative": "Скелетоный ключ",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.key_creative": "Скелетный Ключ",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 3]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_audio": "Звуковой модуль",
+    "item.tis3d.module_bundled_redstone": "Связной редестоун модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_bundled_redstone": "Модуль Красного Многожильного Кабеля",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 4]
```diff
@@ -1,59 +1,59 @@
...
-    "item.tis3d.module_read_only_memory": "ROM Module",
-    "item.tis3d.module_redstone": "Redstone Module",
-    "item.tis3d.module_sequencer": "Sequencer Module",
-    "item.tis3d.module_serial_port": "Serial Port Module",
-    "item.tis3d.module_stack": "Stack Module",
-    "item.tis3d.module_terminal": "Terminal Module",
-    "item.tis3d.module_timer": "Timer Module",
-    "item.tis3d.prism": "Prism",
+    "item.tis3d.book_code": "Код библия",
+    "item.tis3d.book_manual": "TIS-3D Справочное руководство",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.book_manual": "Справочное Руководство TIS-3D",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 5]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.key": "Ключ",
+    "item.tis3d.key_creative": "Скелетоный ключ",
+    "item.tis3d.module_audio": "Звуковой модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_audio": "Звуковой Модуль",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 6]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_bundled_redstone": "Связной редестоун модуль",
+    "item.tis3d.module_display": "Дисплей модуля",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_display": "Дисплейный Модуль",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 7]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_execution": "Выполнененый модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_execution": "Экзекуционный Модуль",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 8]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_infrared": "Инфакрасный модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_infrared": "Инфракрасный Модуль",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 9]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_keypad": "Клавиатурный модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_keypad": "Клавиатурный Модуль",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 10]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_queue": "Модуль очереди",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_queue": "Модуль Очереди",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 11]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_random": "Случайный модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_random": "Модуль Случайности",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 12]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_random_access_memory": "Модуль памяти",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_random_access_memory": "Модуль Оперативной Памяти",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 13]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_read_only_memory": "ПЗУ модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_read_only_memory": "Модуль ПЗУ",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 14]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_redstone": "Редстоун модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_redstone": "Краснопыльный Модуль",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 15]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_sequencer": "Секвенсор модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_sequencer": "Секвенсорный Модуль",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 16]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_serial_port": "Серийный порт модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_serial_port": "Модуль Серийного Порта",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 17]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_stack": "Стак модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_stack": "Стекируемый Модуль",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 18]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_terminal": "Терминал модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_terminal": "Модуль Терминала",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 19]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_timer": "Таймер модуль",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "item.tis3d.module_timer": "Модуль Таймера",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 20]
```diff
@@ -1,59 +1,59 @@
...
-    "tis3d.compiler.invalid_format": "Unexpected token",
-    "tis3d.compiler.invalid_instruction": "Invalid opcode",
-    "tis3d.compiler.label_not_found": "Undefined label",
-    "tis3d.compiler.label_duplicate": "Label is already defined",
-    "tis3d.compiler.parameter_invalid": "Invalid register",
-    "tis3d.compiler.parameter_overflow": "Too many operands",
-    "tis3d.compiler.parameter_underflow": "Missing operand",
-    "tis3d.compiler.too_many_columns": "Line too long",
-    "tis3d.compiler.too_many_lines": "Too many lines",
+    "tis3d.compiler.error": "Ошибка компиляции@%s:%s-%s: ",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.compiler.error": "Ошибка компиляции @%s:%s-%s: ",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 21]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.compiler.invalid_format": "Неожиданный маркер",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.compiler.invalid_format": "Недействительный токен",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 22]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.compiler.invalid_instruction": "Неверный код операции",
+    "tis3d.compiler.label_not_found": "Неопределеная ярлык",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.compiler.label_not_found": "Ярлык не определён",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 23]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.compiler.label_duplicate": "Ярлык уже определен",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.compiler.label_duplicate": "Ярлык уже определён",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 24]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.compiler.parameter_invalid": "Неверный регистр",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.compiler.parameter_invalid": "Неверный реестр",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 25]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.compiler.parameter_overflow": "Слишком много операндов",
+    "tis3d.compiler.parameter_underflow": "Отсутствует операнд",
+    "tis3d.compiler.too_many_columns": "Слишком длинная линия",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.compiler.too_many_columns": "Линия cлишком длинна",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 26]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.compiler.too_many_lines": "Слишком много строк",
 
-    "tis3d.controller.status.incomplete": "Scan failed to complete",
-    "tis3d.controller.status.multiple_controllers": "Multiple controllers",
-    "tis3d.controller.status.too_complex": "Too complex",
+    "tis3d.controller.status.incomplete": "Сканирование не удалось завершить",
+    "tis3d.controller.status.multiple_controllers": "Несколько контроллеров",
+    "tis3d.controller.status.too_complex": "Слишком сложный",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.controller.status.too_complex": "Сложность слишком велика",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 27]
```diff
@@ -1,59 +1,59 @@
...
 
-    "tis3d.manual.home": "Home",
-    "tis3d.manual.blocks": "Blocks",
-    "tis3d.manual.items": "Items",
-    "tis3d.manual.serial_protocols": "Serial Protocols",
-    "tis3d.manual.warning.missing.block": "Block unavailable.",
-    "tis3d.manual.warning.missing.image": "Image not found.",
-    "tis3d.manual.warning.missing.item": "Item unavailable.",
-    "tis3d.manual.warning.missing.tag": "Tag entry unavailable.",
+    "tis3d.manual.home": "Дом",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.manual.home": "Начальная Страница",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 28]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.manual.blocks": "Блокы",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.manual.blocks": "Блоки",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 29]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.manual.items": "Предметы",
+    "tis3d.manual.serial_protocols": "Последовательные протоколы",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.manual.serial_protocols": "Серийные Протоколы",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 30]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.manual.warning.missing.block": "Блокировать недоступно.",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.manual.warning.missing.block": "Блок отсутствует.",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 31]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.manual.warning.missing.image": "Изображение не найдено.",
+    "tis3d.manual.warning.missing.item": "Предмет не доступен.",
+    "tis3d.manual.warning.missing.tag": "Запись тега недуступно.",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.manual.warning.missing.tag": "Таговая запись отсутствует",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 32]
```diff
@@ -1,59 +1,59 @@
...
 
-    "tis3d.book_code.error_on_next_page": "Error on next page:",
-    "tis3d.book_code.error_on_previous_page": "Error on previous page:",
+    "tis3d.book_code.error_on_next_page": "Ошибка на следующей странице:",
+    "tis3d.book_code.error_on_previous_page": "Ошибка на предыдущей странице:",
 
-    "tis3d.tooltip.book_code": "§7Use a book on an installed execution module to get a copy.",
-    "tis3d.tooltip.book_manual": "§7Use a book on a controller block to get a copy.",
-    "tis3d.tooltip.key": "§7Sneak while holding to toggle receiving ports on a casing's face."
+    "tis3d.tooltip.book_code": "§7Используйте книгу об установленном исполнительном модуле, чтобы получить копию.",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.tooltip.book_code": "§7Используй книгу на установленном Экзекуционном Модуле, чтобы получить копию.",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 33]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.tooltip.book_manual": "§7Используйте книгу на блоке контроллера, чтобы получить копию.",
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.tooltip.book_manual": "§7Используй книгу на блоке Контроллера, чтобы получить копию.",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 34]
```diff
@@ -1,59 +1,59 @@
...
+    "tis3d.tooltip.key": "§7Подкрадитесь, удерживая, чтобы переключить приемные порты на лицевой стороне корпуса."
```
Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "tis3d.tooltip.key": "§7Используется для переключения приёмных портов на сторонах Обшивок в режиме скрытности игрока."
```

**Review by Yoghurt4C — CHANGES_REQUESTED** [review]
<a></a>

File: src/main/resources/assets/tis3d/lang/ru_ru.json (hunk 7 above)
Comment (falseresync) [reply src/main/resources/assets/tis3d/lang/ru_ru.json]: Better "Исполняющий Модуль"

File: src/main/resources/assets/tis3d/lang/ru_ru.json (hunk 14 above)
Comment (falseresync) [reply src/main/resources/assets/tis3d/lang/ru_ru.json]: "Редстоуновый Модуль" is more conventional

File: src/main/resources/assets/tis3d/lang/ru_ru.json (hunk 3 above)
Comment (falseresync) [reply src/main/resources/assets/tis3d/lang/ru_ru.json]: A bit unconventional. Mojang calls it "Редстоун". So maybe "Модуль Многожильного Редстоунового Кабеля"

File: src/main/resources/assets/tis3d/lang/ru_ru.json (hunk 17 above)
Comment (falseresync) [reply src/main/resources/assets/tis3d/lang/ru_ru.json]: "Модуль Стека"

File: src/main/resources/assets/tis3d/lang/ru_ru.json [hunk 35]
```diff
@@ -1,59 +1,59 @@
...
+    "item.tis3d.module_serial_port": "Серийный порт модуль",
+    "item.tis3d.module_stack": "Стак модуль",
+    "item.tis3d.module_terminal": "Терминал модуль",
+    "item.tis3d.module_timer": "Таймер модуль",
+    "item.tis3d.prism": "Призма",
 
-    "block.tis3d.casing": "Casing",
-    "block.tis3d.controller": "Controller",
+    "block.tis3d.casing": "Обшивка",
+    "block.tis3d.controller": "Контролер",
```
Comment (falseresync) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]: "Контроллер" - correct spelling

Comment (Yoghurt4C) [inline src/main/resources/assets/tis3d/lang/ru_ru.json]:
```suggestion
    "block.tis3d.controller": "Контроллер",
```

File: src/main/resources/assets/tis3d/lang/ru_ru.json (hunk 31 above)
Comment (falseresync) [reply src/main/resources/assets/tis3d/lang/ru_ru.json]: "Тег отстутствует" - correct spelling + less clunky

File: src/main/resources/assets/tis3d/lang/ru_ru.json (hunk 32 above)
Comment (falseresync) [reply src/main/resources/assets/tis3d/lang/ru_ru.json]: "... Исполняющем Модуле ..."

File: src/main/resources/assets/tis3d/lang/ru_ru.json (hunk 3 above)
Comment (Yoghurt4C) [reply src/main/resources/assets/tis3d/lang/ru_ru.json]: https://minecraft-ru.gamepedia.com/Красная_пыль

**Comment by falseresync** [comment]
@Yoghurt4C check the comments on the suggestions, I didn't use the suggestion tool anywhere :P

File: src/main/resources/assets/tis3d/lang/ru_ru.json (hunk 35 above)
Comment (Sturmlilie) [reply src/main/resources/assets/tis3d/lang/ru_ru.json]: https://en.wiktionary.org/wiki/%D0%BA%D0%BE%D0%BD%D1%82%D1%80%D0%BE%D0%BB%D1%91%D1%80
https://en.wiktionary.org/wiki/%D0%BA%D0%BE%D0%BD%D1%82%D1%80%D0%BE%D0%BB%D0%BB%D0%B5%D1%80
they're two distinct words (I think?), anyway false's suggestion seems more appropriate

Edit: Nvm I thought you were commenting on a suggestion by Yog, but this was just the original translation.

**Pull request closed without merging**